### ISSUE TITLE
All MITRE ATT&CK endpoint should use additional filters

### ISSUE THREAD

**Issue by himynamesdave** [muchdogesec/ctibutler#37]
Any request for a MITRE ATT&CK object should use

```sql
  AND doc.x_mitre_deprecated != true
  AND doc.revoked != true
```

Currently returning revoked and deprecated objects in some requests for objects by id

Ensure these filters are applied on all endpoints

**Comment by fqrious** [comment]
> Currently returning revoked and deprecated objects in some requests for objects by id

Examples?

This is problematic if there's no way to turn it off, revoked attacks are still attacks and users may be interested in the data. I suggest having like a `include_deprecated` filter that defaults to False.

**Comment by himynamesdave** [comment]
@fqrious yea, fair point... although the use-case is rare.

Instead, lets add 

include_deprecated
include_revoked

to all ATT&CK object endpoints GET object // GET objects

**Comment by himynamesdave** [comment]
Needs to exist on 

```
/api/v1/attack-*/objects/{attack_id}/*
```

endpoint too

**Comment by himynamesdave** [comment]
@fqrious you mentioned

> conflicts with _is_latest

can you explain more?

**Comment by fqrious** [comment]
we filter with `_is_latest` by default, this means if a doc is revoked in its latest version, user would not be able to access it until they pass `include_revoked` and/or `include_deprecated`

![Image](https://github.com/user-attachments/assets/97426f72-f75f-4801-9bb0-f909e62c2d35)

Now I think there's an advantage for having this filter in list (/objects/) but none in retrieve (/objects/{id})

**Comment by himynamesdave** [comment]
yes, that's correct. generally if a doc is revoked or deprecated a user does not want to see it, unless there is a very special reason. In short, the owners of the data include all deprecated / revoked objects in the database (these often conflict with existing IDs, e.g. old object revoked, replaced by new one due to errrors)

hence the reason we want to make these default to ignore on all endpoints.

to give you a specific example of why this is a problem on the ID endpoint.

using v16_0

```
curl -X 'GET' \
  'http://127.0.0.1:8006/api/v1/attack-enterprise/objects/T1153/' \
  -H 'accept: application/json'
```

```
{
  "page_size": 50,
  "page_number": 1,
  "page_results_count": 2,
  "total_results_count": 2,
  "objects": [
    {
      "created": "2017-12-14T16:46:06.044Z",
      "created_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
      "description": "**This technique has been deprecated and should no longer be used.**\n\nThe <code>source</code> command loads functions into the current shell or executes files in the current context. This built-in command can be run in two different ways <code>source /path/to/filename [arguments]</code> or <code>.**This technique has been deprecated and should no longer be used.** /path/to/filename [arguments]</code>. Take note of the space after the \".\". Without a space, a new shell is created that runs the program instead of running the program within the current context. This is often used to make certain features or functions available to a shell or to update a specific shell's environment.(Citation: Source Manual)\n\nAdversaries can abuse this functionality to execute programs. The file executed with this technique does not need to be marked executable beforehand.",
      "external_references": [
        {
          "source_name": "mitre-attack",
          "external_id": "T1153",
          "url": "https://attack.mitre.org/techniques/T1153"
        },
        {
          "source_name": "Source Manual",
          "url": "https://ss64.com/bash/source.html",
          "description": "ss64. (n.d.). Source or Dot Operator. Retrieved May 21, 2019."
        }
      ],
      "id": "attack-pattern--45d84c8b-c1e2-474d-a14d-69b5de0a2bc0",
      "kill_chain_phases": [
        {
          "kill_chain_name": "mitre-attack",
          "phase_name": "execution"
        }
      ],
      "modified": "2021-04-29T14:49:39.188Z",
      "name": "Source",
      "object_marking_refs": [
        "marking-definition--fa42a846-8d90-4e51-bc29-71d5b4802168"
      ],
      "spec_version": "2.1",
      "type": "attack-pattern",
      "x_mitre_attack_spec_version": "2.1.0",
      "x_mitre_deprecated": true,
      "x_mitre_detection": "Monitor for command shell execution of source and subsequent processes that are started as a result of being executed by a source command. Adversaries must also drop a file to disk in order to execute it with source, and these files can also detected by file monitoring.",
      "x_mitre_domains": [
        "enterprise-attack"
      ],
      "x_mitre_is_subtechnique": false,
      "x_mitre_modified_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
      "x_mitre_permissions_required": [
        "User"
      ],
      "x_mitre_platforms": [
        "Linux",
        "macOS"
      ],
      "x_mitre_version": "2.0"
    },
    {
      "created": "2018-10-17T00:14:20.652Z",
      "created_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
      "description": "Due to potential legitimate uses of source commands, it's may be difficult to mitigate use of this technique.",
      "external_references": [
        {
          "source_name": "mitre-attack",
          "url": "https://attack.mitre.org/mitigations/T1153",
          "external_id": "T1153"
        }
      ],
      "id": "course-of-action--5391ece4-8866-415d-9b5e-8dc5944f612a",
      "modified": "2019-07-25T11:45:45.651Z",
      "name": "Source Mitigation",
      "object_marking_refs": [
        "marking-definition--fa42a846-8d90-4e51-bc29-71d5b4802168"
      ],
      "spec_version": "2.1",
      "type": "course-of-action",
      "x_mitre_attack_spec_version": "2.1.0",
      "x_mitre_deprecated": true,
      "x_mitre_domains": [
        "enterprise-attack"
      ],
      "x_mitre_modified_by_ref": "identity--c78cb6e5-0c4b-4611-8297-d1b8b55e40b5",
      "x_mitre_version": "1.0"
    }
  ]
}
```

The course-of-action is deprecated, and user does not want this (usually). Hence, why we should ignore by default. However, include the two url filters so that user can choose to include it, if their use-case requires it (unlikely)

So in short, this filter is in addition to is latest, it does not replace it.
